src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordObjEquals.cs

```diff
@@ -120,13 +124,32 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
         {
             var F = new SyntheticBoundNodeFactory(this, ContainingType.GetNonNullSyntaxNode(), compilationState, diagnostics);
 
-            // return this.Equals(param as ContainingType);
-            var block = F.Block(ImmutableArray.Create<BoundStatement>(
-                F.Return(
-                    F.InstanceCall(F.This(), "Equals",
-                        F.As(F.Parameter(Parameters[0]), ContainingType)))));
-
-            F.CloseMethod(block);
+            var paramAccess = F.Parameter(Parameters[0]);
+
+            BoundExpression expression;
+            if (ContainingType.IsStructType())
+            {
+                // For structs:
+                //
+                //      return param is ContainingType i ? this.Equals(in i) : false;
+                expression = F.Conditional(
+                    F.Is(paramAccess, ContainingType),
+                    F.Call(
+                        F.This(),
+                        _typedRecordEquals,
+                        ImmutableArray.Create<RefKind>(RefKind.In),
```

Comment:
This isn't really `param is ContainingType i ? this.Equals(in i) : false`, it's `param is ContainingType ? this.Equals(in (ContainingType)i) : false`, right? Unless I'm missing something here?

Reply:
Yeah, but those are basically the same thing.